lib/collection/src/shards/mod.rs

```diff
@@ -41,6 +41,11 @@ pub fn shard_path(collection_path: &Path, shard_id: ShardId) -> PathBuf {
     collection_path.join(format!("{shard_id}"))
 }
 
+/// Path to a shard directory
+pub fn shard_initialized_flag_path(collection_path: &Path, shard_id: ShardId) -> PathBuf {
+    collection_path.join(format!("shard_{shard_id}.initialized"))
```

Comment:
```suggestion
    collection_path.join(format!("shard_{shard_id}.initializing"))
```

To indicate that the file is temporary